### ISSUE TITLE
How to make config dynamic

### ISSUE THREAD

**Issue by timsatterfield** [devise-security/devise-security#62]
Wondering how to best make the config attributes dynamic per instance of a model.

Example: for a given Model which belongs to a specific Organization might have different password validation requirements or password expiry length.

Overriding password_complexity, password_length or other methods in the Model doesn't seem to do anything. Any help here?

**Comment by olbrich** [comment]
Can you give me a made up code example of what you would like to be able to do?

**Comment by timsatterfield** [comment]
```ruby
class Manager < ActiveRecord::Base
  devise :database_authenticatable, :registerable,
         :recoverable, :trackable,
         :confirmable, :lockable,
         :secure_validatable

  belongs_to :team

  # Override password_length enforced by secure_validatable
  def password_length
  	team.password_length | # return value from the devise config file
  end

  def password_complexity
  	team.password_complexity | # return value from the devise config file
  end

  # Have similar overrides for other module rules etc

end
```

**Comment by olbrich** [comment]
On the surface this seems like it could work, but I suspect what is happening is that those methods are being evaluated at the time the class is loaded and never again after that.  I'll dig into it some more and see if we can verify this and modify this so that it can be a proc or something.

**Comment by rmorlok** [comment]
Is there any way that something like this could make its way into the gem? I have a similar requirement where I have customers with sets of users that have differing IT policies on things like password complexity requirements, etc.

**Comment by olbrich** [comment]
@rmorlok yes, I'll update #63 and see if we can get this working again.

**Comment by AndrewJGregory** [comment]
Any updates on this?

**Comment by khan-arkhitech-2020** [comment]
Hey guyz whats the status for this task

**Comment by WilfredRuck** [comment]
Hey, I could also make use of this. Any updates or in between solutions available?

**Comment by dillonwelch** [comment]
This will be released in `0.17.0` shortly.
